CMakeLists.txt

```diff
@@ -1,5 +1,13 @@
 # Defines Taraxa library target.
 
+if (DEFINED SYSTEM_HOME_OVERRIDE)
+    list(APPEND CMAKE_PREFIX_PATH ${SYSTEM_HOME_OVERRIDE})
+    include_directories(${SYSTEM_HOME_OVERRIDE}/include)
+    link_directories(${SYSTEM_HOME_OVERRIDE}/lib)
+endif ()
```

Comment:
This was the easiest way to support submodule builds (some of them need custom installation of openssl and boost on mac) + support FindXXX on the same custom install dir

Reply:
I guess I am ok with this. Just want to point out that there is a direct support in cmake for overriding boost and openssl custom instalation, not sure if that is the only challenge here, but if yes we can use cmake arguments: 
-DBOOST_ROOT=/path/to/custom/boost... -> [example](https://stackoverflow.com/questions/38905480/how-to-pass-boost-root-and-boost-library-to-cmake) [cmake doc](https://cmake.org/cmake/help/latest/module/FindBoost.html)
-DOPENSSL_ROOT_DIR=/path/to/custom/openssl...  -> example follows same logis as boost example, [cmake doc](https://cmake.org/cmake/help/v3.6/module/FindOpenSSL.html)

Reply:
We also need a way to provide custom clang-format and cppcheck. This is a generic solution, should cover many potential needs. And only needs setting single config variable.